### PR TITLE
Ratom 13 data modeling file account

### PR DESCRIPTION
**Pull request by JeremyGibson**
I think this is a good start on where we will need to be at the end of the project. Most of this work is in the models file.  I had to remove some thing from other files to allow migrations to run.

NOTE: This is a complete change from the original modeling so if you still need to run with an imported account don't apply this branch until the importer is ready.

Adds: an exclusion to pre-commit so that migrations aren't blacked

### CODE REVIEW

File: ratom/models.py [hunk 1]
```diff
@@ -30,48 +74,106 @@ class Processor(models.Model):
     )
 
 
+class RestrictionAuthority(models.Model):
+    authorities = ArrayField(base_field=models.CharField(max_length=128, blank=True))
+
+
+class Redaction(models.Model):
+    redacted_subject = models.TextField(null=True)
```
Comment (copelco) [inline ratom/models.py]: swap to ``blank=True``

File: ratom/models.py [hunk 2]
```diff
@@ -30,48 +74,106 @@ class Processor(models.Model):
...
+    redacted_body = models.TextField(null=True)
```
Comment (copelco) [inline ratom/models.py]: swap to ``blank=True``

File: ratom/models.py [hunk 3]
```diff
@@ -30,48 +74,106 @@ class Processor(models.Model):
...
+    account = models.ForeignKey(Account, on_delete=models.PROTECT)
+    restrictions = models.ForeignKey(
+        RestrictionAuthority, null=True, on_delete=models.CASCADE
     )
-    sent_date = models.DateTimeField()
-    msg_from = models.TextField()
-    msg_to = models.TextField()
+    redaction = models.ForeignKey(Redaction, null=True, on_delete=models.CASCADE)
+    sent_date = models.DateTimeField(null=True)
+    msg_from = models.TextField(null=True)
```
Comment (copelco) [inline ratom/models.py]: swap to ``blank=True``

File: ratom/models.py [hunk 4]
```diff
@@ -30,48 +74,106 @@ class Processor(models.Model):
...
+    msg_to = models.TextField(null=True)
```
Comment (copelco) [inline ratom/models.py]: swap to ``blank=True``

File: ratom/models.py [hunk 5]
```diff
@@ -1,24 +1,68 @@
...
+    filename = models.CharField(max_length=200)
+    reported_total_messages = models.IntegerField(null=True)
+    accession_date = models.DateField(null=True)
+    file_size = models.IntegerField(null=True)
+    md5_hash = models.CharField(max_length=32)
+    import_status = models.CharField(
+        max_length=32,
+        choices=[(tag, tag.value) for tag in FileImportStatus],
+        default=FileImportStatus.CREATED,
+    )
```
Comment (copelco) [inline ratom/models.py]: Add auto_now_add DateTime field ``date_imported``

File: ratom/models.py [hunk 6]
```diff
@@ -30,48 +74,106 @@ class Processor(models.Model):
...
+                }],
+            },
+            errors: [{
+                int: string,
+            }],
+            raw: string ## Text dump of an errored message,
+        }
+    """
+
+    source_id = models.CharField(max_length=256, blank=True)
```
Comment (copelco) [inline ratom/models.py]: Should we allow empty values for this?

**Comment by JeremyGibson** [comment]
This should be working with the DRF-ication. I haven't tested it with the front end yet. But it does index.

File: core/models.py [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+from django.contrib.postgres.fields import JSONField, ArrayField
+from django.db import models
+
+from simple_history.models import HistoricalRecords
+from elasticsearch_dsl import Index
+from django_elasticsearch_dsl_drf.wrappers import dict_to_obj
+
+
+class FileImportStatus(Enum):
+    CREATED = "Created"
```
Comment (copelco) [inline core/models.py]: I don't think we're using these enums properly.

File: deployment/ci_deployment.sh [hunk 8]
```diff
@@ -29,4 +29,4 @@ fi
 
 echo $VAULT_PASS > .vault_pass
 ansible-galaxy install -r requirements.yaml
-ansible-playbook deploy.yaml -l ratom-staging -e k8s_container_image_tag=$TAG
+ansible-playbook deploy.yaml -l api-staging -e k8s_container_image_tag=$TAG
```
Comment (copelco) [inline deployment/ci_deployment.sh]:
```suggestion
ansible-playbook deploy.yaml -l ratom-staging -e k8s_container_image_tag=$TAG
```

File: deployment/inventory.yaml [hunk 9]
```diff
@@ -8,10 +8,10 @@ all:
           env_name: staging
           k8s_kube_context: microk8s
           k8s_domain_names:
-          - ratom-staging.caktus-built.com
-          k8s_namespace: "ratom-{{ env_name }}"
+          - api-staging.caktus-built.com
```
Comment (copelco) [inline deployment/inventory.yaml]: These should all stay as ``ratom``

File: flush.sql [hunk 10]
```diff
@@ -0,0 +1,15 @@
+BEGIN;
```
Comment (copelco) [inline flush.sql]: Is this file used anywhere?

**Review by copelco — CHANGES_REQUESTED** [review]
Found a few stray search/replace issues

File: flush.sql (hunk 10 above)
Comment (JeremyGibson) [reply flush.sql]: Sorry, not sure how that made into the repo. No that's something I use personally.

File: deployment/inventory.yaml (hunk 9 above)
Comment (copelco) [reply deployment/inventory.yaml]: @JeremyGibson I think all of the ``api-`` additions in this file need to revert back to ``ratom-``

Comment (JeremyGibson) [reply deployment/inventory.yaml]: I left the kubernetes ingress as api

**Review by copelco — APPROVED** [review]

**Pull request merged**